Scripts/RemediationScripts/Remediate-CdnFrontDoorMinTLSVersion.ps1

```diff
@@ -0,0 +1,1062 @@
+            Write-Host $([Constants]::SingleDashLine)
+            return
+        }
+        Write-Host "User has provided consent to reset minimum TLS Version on all Front Doors." -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host $([Constants]::SingleDashLine)
+    }
+    else
+    {
+        Write-Host "'Force' flag is provided. Minimum TLS Version will be reset on all Front Doors without any further prompts." -ForegroundColor $([Constants]::MessageType.Warning) -NoNewline
+        Write-Host $([Constants]::SingleDashLine)
+    }
+
+    Write-Host "[Step 3 of 3] Reset Minimum TLS Version for Front Doors" 
+    Write-Host $([Constants]::SingleDashLine)
+
+    # Includes Front Doors, to which, previously made changes were successfully rolled back.
+    $frontDoorsRolledBack = @()
+
+    # Includes Front Doors that were skipped during roll back. There were errors rolling back the changes made previously.
+    $frontDoorsEPSkipped = @()
+
+    #$requiredMinTLSVersion = 'TLS10'
+
+    $validFrontDoorEndpointDetails | ForEach-Object {
+        $frontdoorEndpoint = $_
+        $resourceGroupName = $_.ResourceGroupName
+        $resourceName = $_.ResourceName
+        $DomainName = $_.DomainName
+        $CertificateType =$_.CertificateType
+        $minimumTlsVersion = $_.MinimumTlsVersion
+        $prevMinimumTlsVersion =$_.PrevMinimumTlsVersion
+        $isMinTLSVersionSetOnCustomDomain = $_.isMinTLSVersionSetOnCustomDomain
+        $isMinTLSVersionRolledBackCustomDomainPostRemediation = $false;
+
+        $frontdoorEndpoint | Add-Member -NotePropertyName isMinTLSVersionRolledBackCustomDomainPostRemediation -NotePropertyValue $isMinTLSVersionRolledBackCustomDomainPostRemediation
+        
+        try
+        {
+            Write-Host "Fetching Front Door configuration: Resource Group Name: [$($resourceGroupName)], Resource Name: [$($resourceName)]..." -ForegroundColor $([Constants]::MessageType.Info)
+            Write-Host $([Constants]::SingleDashLine)
+
+                $secrets =  Get-AzFrontDoorCdnSecret -ResourceGroupName $resourceGroupName -ProfileName $resourceName
+                $secretObj = $null
+
+                foreach($secret in $secrets)
+                {
+                    if($secret.Name.Contains($DomainName))
+                    {
+                        $secretObj = $secret
+                    }
+                }
+                $secretResoure = New-AzFrontDoorCdnResourceReferenceObject -Id $secretObj.Id
+                $updateTlsSetting = New-AzFrontDoorCdnCustomDomainTlsSettingParametersObject -CertificateType $CertificateType -MinimumTlsVersion $prevMinimumTlsVersion -Secret $secretResoure
+                $resource = Update-AzFrontDoorCdnCustomDomain -ResourceGroupName $resourceGroupName -ProfileName $resourceName -CustomDomainName $DomainName -TlsSetting $updateTlsSetting
+                    
+                if($null -ne $resource -and $updateTlsSetting.MinimumTlsVersion -eq 'TLS10' )
```

Comment:
Instead of comparing with hard coded value 'TLS10', we can compare with previous value stored. So that if in future TLS11 is also supported then no change in script will be needed.

Reply:
Added prev min tls version